### PR TITLE
Make commands re-orderable w/ drag+drop, add context menu

### PR DESCRIPTION
**Pull request by WiIIiam278**
Still WIP -- need to make context menu things actually update the list

### CODE REVIEW

**Review by jonko0493 — COMMENTED** [review]
I think this looks great, but I have a few questions that I want to know the answers to before we merge it.

File: src/SerialLoops.Lib/Script/ScriptItemCommand.cs [hunk 1]
```diff
@@ -629,5 +631,17 @@ private static DialogueLine GetDialogueLine(short index, EventFile eventFile)
...
         }
+
+        public ScriptItemCommand Clone()
+        {
+            return new()
+            {
+                Verb = Verb,
+                Parameters = Parameters.Select(p => p.Clone()).ToList(),
+                Section = Section,
+                Index = Index,
```
[the remark this answers is not in the document]
Comment (WiIIiam278) [reply src/SerialLoops.Lib/Script/ScriptItemCommand.cs:642]: yes it absolutely should, uhh, I wasn't entirely familiar with the complete internals of how that worked on the lib side, so I left it as is for now

File: src/SerialLoops/Controls/ContextMenu.cs [hunk 2]
```diff
@@ -97,4 +97,92 @@ private void ExportNames_OnClick(object sender, EventArgs e)
...
+
+        private void OnPasteItem(object sender, EventArgs e)
+        {
+            if (_treeView.SelectedCommandTreeItem is null) return;
+            if (_clipboard is null) return;
+            if (_clipboardIsCut)
+            {
+                _treeView.DeleteItem(_clipboard);
+            }
+            _treeView.AddItem(_clipboard);
```
[the remark this answers is not in the document]
Comment (WiIIiam278) [reply src/SerialLoops/Controls/ContextMenu.cs:167]: Inside the AddItem method itself it checks the users cursor position inside the tree and does an insert into the list at an index +1 of that

File: src/SerialLoops.Lib/Script/ScriptItemCommand.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (jonko0493) [reply src/SerialLoops.Lib/Script/ScriptItemCommand.cs:640]: Got it, will fill in when I take over after this one.

File: src/SerialLoops/Controls/ContextMenu.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (jonko0493) [reply src/SerialLoops/Controls/ContextMenu.cs:167]: Sick, sounds reasonable.

**Review by jonko0493 — APPROVED** [review]
I'd say you're good to merge after you at least root cause why the cloned commands can't be deleted. Fantastic work! 😄

File: src/SerialLoops/Controls/ContextMenu.cs [hunk 3]
```diff
@@ -97,4 +97,92 @@ private void ExportNames_OnClick(object sender, EventArgs e)
...
+            cutCommand.Executed += OnCutItem;
+            copyCommand.Executed += OnCopyItem;
+            pasteCommand.Executed += OnPasteItem;
+            deleteCommand.Executed += OnDeleteItem;
+        }
+
+        private void OnDeleteItem(object sender, EventArgs e)
+        {
+            if (_treeView.SelectedCommandTreeItem is null) return;
+            _treeView.DeleteItem(_treeView.SelectedCommandTreeItem);
```
[the remark this answers is not in the document]
Comment (WiIIiam278) [reply src/SerialLoops/Controls/ContextMenu.cs:156]: Fixed this, turns out I wasn't setting the parent of the inserted item correctly :)

**Pull request merged**